### ISSUE TITLE
Allow users to set their private "github"  or check multiple urls for github dependencies

### ISSUE THREAD

**Issue by coltonw** [npm/npm#17838]
#### What's the feature?
Have a local npm configuration for `npm config set github` (similar to npm set registry).  This would allow:
```json
{
  "name": "foo",
  "version": "0.0.0",
  "dependencies": {
    "coltonw/awesomelib"
  }
}
```
to potentially be pulling from github.example.com/coltonw/awesomelib rather than github.com/coltonw/awesomelib.

Alternatively, you could use something like scoped registries to scope githubs?  Like `npm config set @mycompany:github github.mycompany.com`. That way you can do the following:
```json
{
  "name": "foo",
  "version": "0.0.0",
  "dependencies": {
    "@mycompany/coltonw/awesomelib"
  }
}
```
You could actually do this in addition to the first option as well.

Another alternative, allow multiple git urls for one dependency.

#### What problem is the feature intended to solve?
We have two teams in different networks, each with their own private Github Enterprise instance. We have a repository with git dependencies, and by necessity the parent repo and the dependency repos must all be mirrored with remotes on both github instances. In order to use git URLs for these dependencies, we basically need to have lines of the package.json file differ between the two mirrored repositories on each of the github instances, which git is really not good at doing if you are ever merging those repos.

If there is already a better way to solve this, I would be overjoyed to hear it.

#### Is the absence of this feature blocking you or your team? If so, how?
It is a big impediment because right now we need to manually edit package.json every time we merge between their remote git repo and our remote git repo.  This limits who can merge between our two repos to only users who understand this quirk very well.

#### Is this feature similar to an existing feature in another tool?
It is similar to a feature already existing in this tool of setting another registry.  Pretty much all dependency managers have a way to add alternative "resolvers".

I do not know of a tool that allows a backup URL for a specific dependency.

#### Is this a feature you're prepared to implement, with support from the npm CLI team?
Depending on scope, yes.

**Comment by maciek134** [comment]
Did you find another way of solving this problem?

**Comment by coltonw** [comment]
Nope.  We do what I implied in the "What problem is the feature intended to solve?" section: whenever we merge one repo into the other, we have a follow-up commit fixing the package.json to continue pointing to the correct github instance.
